### PR TITLE
Final clean up note features

### PR DESCRIPTION
**Pull request by xSabrina**
**Description**
Concerns #152 

This changes contain clean ups mainly from note-related classes. 
This means in detail:

- some blank lines were inserted for better readability.
- getter methods (like getContext()) have been replaced by require ones wherever possible to avoid NullPointerExceptions.
- some code has been simplified (e.g. by replacing getArguments() by requireArguments() and remove the line for avoiding NullPointerExceptions (getArguments != null).
- comments have been changed in format, content or type (all method comments are JavaDoc comments now). 
- some codelines have been reordered.
- imports and class variables have been optimized in sorting.
- a method has been simplified and shortened by outsourcing the code partially (new playOrPause() method).
- code has been simplified by declaring the mainActivity as context once and reuse the class variable instead of calling the requireContext() or requireActivity() (or according getter methods) multiple times within a class.

Also the gradle has been updated by declaring the newest versions of some dependencies that had been missing or were older.

**Affects**
Mainly note-related classes and the gradle.

**Notes for Reviewer**
Please review & comment/approve. Thanks!

### CODE REVIEW

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java [hunk 1]
```diff
@@ -25,10 +25,10 @@
 import java.util.Locale;
 
 /**
- * NoteRecyclerViewAdapter provides a binding from the noteList to the view
- * that is displayed within the RecyclerView of the NotesFragment.
+ * NoteRecyclerViewAdapter provides a binding from a note-list to a corresponding
+ * RecyclerView-list.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java:29]: As far as I now, in this case in the english language no "-" would be used :)
```suggestion
 * NoteRecyclerViewAdapter provides a binding from a note list to a corresponding
 * RecyclerView list.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java [hunk 2]
```diff
@@ -42,17 +42,18 @@
...
   /**
-   * Adapter constructor to connect a NoteList with the activity.
+   * Constructor to connect a NoteList with a MainActivity.
    *
-   * @param activity  Base activity
-   * @param noteList  The note list.
-   * @param noteModel The note model.
+   * @param activity  instance of MainActivity.
+   * @param noteList  of NoteItems.
+   * @param noteModel model for handling Note-objects.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java:56]:
```suggestion
   * @param noteModel model for handling Note objects.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java [hunk 3]
```diff
@@ -73,19 +74,17 @@ public NotesViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     return new NotesViewHolder(itemView);
   }
 
-
   /**
-   * Method to setup the custom ViewHolder components for notes.
+   * Method to set up custom ViewHolder components for Notes.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java:78]: In line with the suggestions you gave me in #169 I would advise to use active wording here.
```suggestion
   * Set up custom ViewHolder components for Notes.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java [hunk 4]
```diff
@@ -73,19 +74,17 @@ public NotesViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
    *
-   * @param holder   custom ViewHolder instance
-   * @param position adapterPosition of the viewHolder-item
+   * @param holder   custom ViewHolder instance.
+   * @param position within the adapter for the viewHolder-item.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java:81]:
```suggestion
   * @param position within the adapter for the viewHolder item.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java [hunk 5]
```diff
@@ -220,6 +204,29 @@ private void setupMediaPlayerListeners(MediaPlayer mediaPlayer, SeekBarCompat pr
...
+      } else {
+        mediaPlayer.pause();
+        button.setImageResource(R.drawable.play);
+      }
+      paused = !paused;
+    } else {
+      resetPlayers();
+      startAudio(mediaPlayer, button, noteItem, seekBarListener);
+      stopButton.setClickable(true);
+    }
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java:226]: I would add one or two new lines in this area to structure the code a bit :)

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java [hunk 6]
```diff
@@ -301,12 +310,13 @@ public void setNoteList(List<NoteItem> noteList) {
     private final TextView totalTime;
 
     /**
-     * Custom ViewHolder constructor to setup its basic view.
+     * Constructor to set up the Note-CardView.
      *
-     * @param itemView View of the RecyclerView-item.
+     * @param itemView view of the corresponding RecyclerView-item.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java:315]:
```suggestion
     * @param itemView view of the corresponding RecyclerView item.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java [hunk 7]
```diff
@@ -301,12 +310,13 @@ public void setNoteList(List<NoteItem> noteList) {
     private final TextView totalTime;
 
     /**
-     * Custom ViewHolder constructor to setup its basic view.
+     * Constructor to set up the Note-CardView.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java:313]:
```suggestion
     * Constructor to set up the Note CardView.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 8]
```diff
@@ -242,9 +260,15 @@ public void bullet(boolean valid) {
     }
   }
 
-  //Check if the line does not contain a given span-Object yet and apply it, if that is true.
+  /**
+   * Check if the line does not contain a given span-Object yet and apply it, if that is true.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:264]:
```suggestion
   * Check if the line does not contain a given span object yet and apply it, if that is true.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 9]
```diff
@@ -242,9 +260,15 @@ public void bullet(boolean valid) {
...
+   *
+   * @param contain boolean if the given span-Object is already contained in the relevant string.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:266]:
```suggestion
   * @param contain boolean if the given span object is already contained in the relevant string.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 10]
```diff
@@ -297,34 +333,49 @@ private void adjustCursor(int lineStart, int lineEnd) {
     }
   }
 
-  //Apply a given span-Object to a text line.
+  /**
+   * Apply a given span-Object to a text line.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:337]:
```suggestion
   * Apply a given span object to a text line.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 11]
```diff
@@ -297,34 +333,49 @@ private void adjustCursor(int lineStart, int lineEnd) {
...
+              new BackgroundColorSpan(ContextCompat.getColor(context, R.color.gray)),
               start, end, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
         }
       }
     }
   }
 
-  //Check if the line does contain a given span-Object yet and remove it, if that is true.
+  /**
+   * Check if the line does contain a given span-Object yet and remove it, if that is true.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:369]:
```suggestion
   * Check if the line does contain a given span object yet and remove it, if that is true.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 12]
```diff
@@ -297,34 +333,49 @@ private void adjustCursor(int lineStart, int lineEnd) {
...
+   *
+   * @param contain boolean if the given span-object is already contained in the string.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:371]:
```suggestion
   * @param contain boolean if the given span object is already contained in the string.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 13]
```diff
@@ -334,21 +385,31 @@ private void checkContaining(Boolean contain, Object span) {
         lineEnd = lineStart + lines[i].length();
         removeLineSpan(lineStart, lineEnd, span);
       }
+
     }
   }
 
-  //Remove a given span-object from a text line.
+  /**
+   * Remove a given span-object from a text line.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:393]:
```suggestion
   * Remove a given span object from a text line.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 14]
```diff
@@ -362,33 +423,44 @@ private void removeLineSpan(int lineStart, int stringLength, Object span) {
...
+
           removeStyleSpan(FORMAT_ITALIC, start, end);
         }
       }
     }
   }
 
-  //Return whether the editable text contains a given span-object.
+  /**
+   * Return whether the editable text contains a given span-object.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:440]:
```suggestion
   * Return whether the editable text contains a given span object.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 15]
```diff
@@ -397,20 +469,29 @@ private boolean lineContainsFormat(Object span) {
...
     } while (lineContainsFormat(i, span));
+
     return false;
   }
 
-  //Return whether a line at given index from the entire editable text lines contains a given
-  // span-object yet.
+  /**
+   * Return whether a line at given index from the entire editable text lines contains a given
+   * span-object yet.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:478]:
```suggestion
   * span object yet.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java [hunk 16]
```diff
@@ -397,20 +469,29 @@ private boolean lineContainsFormat(Object span) {
...
+   *
+   * @param index of the line.
+   * @param span  object that should be checked for.
+   * @return true if the line already contains the given span-object.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/RichTextEditor.java:482]:
```suggestion
   * @return true if the line already contains the given span object.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/TextNoteEditorFragment.java [hunk 17]
```diff
@@ -49,7 +50,7 @@ protected void onBackPressed() {
   public void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
 
-    MainActivity mainActivity = (MainActivity) requireActivity();
+    mainActivity  = (MainActivity) requireActivity();
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/TextNoteEditorFragment.java:53]:
```suggestion
    mainActivity = (MainActivity) requireActivity();
```

File: BibBuddy/app/src/main/java/de/bibbuddy/TextNoteEditorFragment.java [hunk 18]
```diff
@@ -66,29 +67,34 @@ public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
...
-      ((MainActivity) requireActivity()).openImprint();
+      mainActivity.openImprint();
     }
+
     return super.onOptionsItemSelected(item);
   }
 
-  //Show the TextNoteEditorFragment help-element.
+  /**
+   * Show the TextNoteEditorFragment help-element.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/TextNoteEditorFragment.java:81]:
```suggestion
   * Show the TextNoteEditorFragment help element.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/TextNoteEditorFragment.java [hunk 19]
```diff
@@ -66,29 +67,34 @@ public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
...
+    HelpFragment helpFragment = new HelpFragment();
+    helpFragment.setArguments(bundle);
     helpFragment
-        .show(requireActivity().getSupportFragmentManager(), LibraryKeys.FRAGMENT_HELP_VIEW);
+        .show(mainActivity.getSupportFragmentManager(), LibraryKeys.FRAGMENT_HELP_VIEW);
   }
 
-  //Save the current text as note-object.
+  /**
+   * Save the current text as note-object.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/TextNoteEditorFragment.java:96]:
```suggestion
   * Save the current text as Note object.
```

File: BibBuddy/app/src/main/java/de/bibbuddy/VoiceNoteEditorFragment.java [hunk 20]
```diff
@@ -94,12 +91,14 @@ public boolean onOptionsItemSelected(MenuItem item) {
     return super.onOptionsItemSelected(item);
   }
 
-  //Show the VoiceNoteEditorFragment help-element.
+  /**
+   * Show the VoiceNoteEditorFragment help-element.
```
Comment (LuisMossburger) [inline BibBuddy/app/src/main/java/de/bibbuddy/VoiceNoteEditorFragment.java:95]:
```suggestion
   * Show the VoiceNoteEditorFragment help element.
```

**Review by LuisMossburger — CHANGES_REQUESTED** [review]
Hey, thanks for letting me review! Looks good to me, only some minor suggestions regarding the wording :) Other than that, I'm in!

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java (hunk 5 above)
Comment (xSabrina) [reply BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java]: You are right - this is done.

**Review by LuisMossburger — APPROVED** [review]
Look's good to me now, thank you very much! :)

**Pull request merged**